apps/api/src/applicant/ienapplicant.service.ts

```diff
@@ -239,12 +244,12 @@ export class IENApplicantService {
         throw new BadRequestException('Provided applicant and competition/job does not match');
       }
     }
-    if (!job) {
+    if (data.status.parent?.id === 10003 && !job) {
       throw new BadRequestException(`Competition/job are required to add a milestone`);
     }
 
-    if (user_id) {
-      const added_by_data = await this.ienUsersRepository.findOne(user_id);
+    if (user.user_id) {
+      const added_by_data = await this.ienUsersRepository.findOne(user.user_id);
```

Comment:
Can we just set user_id? `data.added_by = user.user_id` instead of querying user.

Reply:
the query returns a User object containing name, email etc.  not just assigning the id to the field

Reply:
it doesn't return `data` object but passes it to addApplicantStatusAudit method. My understanding is that we don't need to resolve relation when adding an entry but addApplicantStatusAudit should do.